module/documents/activity/transform.mjs

```diff
@@ -0,0 +1,188 @@
+import TransformSheet from "../../applications/activity/transform-sheet.mjs";
+import TransformUsageDialog from "../../applications/activity/transform-usage-dialog.mjs";
+import CompendiumBrowser from "../../applications/compendium-browser.mjs";
+import TransformActivityData from "../../data/activity/transform-data.mjs";
+import { getSceneTargets, simplifyBonus } from "../../utils.mjs";
+import ActivityMixin from "./mixin.mjs";
+
+/**
+ * Activity for transforming an actor into something else.
+ */
+export default class TransformActivity extends ActivityMixin(TransformActivityData) {
+  /* -------------------------------------------- */
+  /*  Model Configuration                         */
+  /* -------------------------------------------- */
+
+  /** @inheritDoc */
+  static LOCALIZATION_PREFIXES = [...super.LOCALIZATION_PREFIXES, "DND5E.TRANSFORM"];
+
+  /* -------------------------------------------- */
+
+  /** @inheritDoc */
+  static metadata = Object.freeze(
+    foundry.utils.mergeObject(super.metadata, {
+      type: "transform",
+      img: "systems/dnd5e/icons/svg/activity/transform.svg",
+      title: "DND5E.TRANSFORM.Title",
+      sheetClass: TransformSheet,
+      usage: {
+        actions: {
+          transformActor: TransformActivity.#transformActor
+        },
+        dialog: TransformUsageDialog
+      }
+    }, { inplace: false })
+  );
+
+  /* -------------------------------------------- */
+
+  /** @inheritDoc */
+  static localize() {
+    super.localize();
+    this._localizeSchema(this.schema.fields.profiles.element, ["DND5E.TRANSFORM.FIELDS.profiles"]);
+  }
+
+  /* -------------------------------------------- */
+  /*  Properties                                  */
+  /* -------------------------------------------- */
+
+  /**
+   * Does the user have permissions to transform?
+   * @type {boolean}
+   */
+  get canTransform() {
+    return game.user.can("ACTOR_CREATE") && (game.user.isGM || game.settings.get("dnd5e", "allowPolymorphing"));
+  }
+
+  /* -------------------------------------------- */
+  /*  Activation                                  */
+  /* -------------------------------------------- */
+
+  /**
+   * @typedef {ActivityUseConfiguration} TransformUseConfiguration
+   * @property {Partial<TransformationConfiguration>} summons  Options for configuring transformation behavior.
+   */
+
+  /**
+   * @typedef TransformationConfiguration
+   * @property {string} profile  ID of the transformation profile to use.
+   * @property {string} [uuid]   UUID of the creature to transform into.
+   */
+
+  /* -------------------------------------------- */
+
+  /** @inheritDoc */
+  _prepareUsageConfig(config) {
+    config = super._prepareUsageConfig(config);
+    config.transform ??= {};
```

Comment:
Missing `create.transform`.

Reply:
`create.transform` isn't part of the configuration because transformation buttons are always added to the message. This is similar to how the Enchant activity is set up (the usage dialog still uses `create.transform` when checking `_sourceDisplay`, which matches Enchant which also has a profile but not checkbox).

Reply:
How does this part work then?

> When the transform activity is used the usage dialog will appear if more than one valid profile is present.

I guess it instead needs a `_requiresConfigurationDialog` override, like `EnchantActivity`?

Reply:
That would indeed be the case 👍